crates/astria-account-monitor/src/account_monitor.rs

```diff
@@ -82,40 +98,70 @@ impl AccountMonitor {
     ///
     /// # Errors
     /// An error is returned if bridge last transaction height is not found.
-    pub async fn run(&self) -> eyre::Result<()> {
-        let shutdown_token = ShutdownHandle::new();
-        let Some(()) = shutdown_token
-            .token
-            .run_until_cancelled(run_loop(
+    async fn run(self) -> eyre::Result<()> {
+        let mut poll_timer = interval(self.interval);
+        poll_timer.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
+
+        loop {
+            if self.shutdown_token.is_cancelled() {
```

Comment:
You have brought back the behaviour we already fixed in the meantime. Use `CancellationToken::run_until_cancelled` or use `tokio::select!` with a `CancellationToken::cancelled` arm.